src/elements/Container/Container.js

```diff
@@ -1,33 +1,68 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
-import META from '../../utils/Meta'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+import META from '../../utils/Meta';
+import { getUnhandledProps, useKeyOnly } from '../../utils/propUtils'
 
 /**
- * A container that gives your content some side padding.
+ * A container limits content to a maximum width
  */
-export default class Container extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-  }
-
-  static _meta = {
-    library: META.library.semanticUI,
-    name: 'Container',
-    type: META.type.element,
-  }
-
-  render() {
-    const classes = classNames(
-      'sd-container',
-      'ui',
-      this.props.className,
-      'container',
-    )
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+function Container(props) {
+  const {
+    text, left, center, right, justified, fluid,
+    children, className,
+  } = props
+
+  const classes = cx('sd-container ui',
+    useKeyOnly(text, 'text'),
+    useKeyOnly(left, 'left aligned'),
+    useKeyOnly(center, 'center aligned'),
+    useKeyOnly(right, 'right aligned'),
+    useKeyOnly(justified, 'justified'),
+    useKeyOnly(fluid, 'fluid'),
+    'container',
+    className
+  )
+
+  const ContainerComponent = 'div'
+  const rest = getUnhandledProps(Container, props)
+
+  return (
+    <ContainerComponent className={classes} {...rest}>
```

Comment:
Absolutely.  Wasn't sure how stringently to follow your initial efforts 
on Label.  Will optimize as the default cause of action in future.

> Levi Thomason <mailto:notifications@github.com>
> 24 June 2016 at 17:24
>
> In src/elements/Container/Container.js 
> <https://github.com/TechnologyAdvice/stardust/pull/277#discussion_r68423440>:
>
> > + const classes = cx('sd-container ui',
> > + useKeyOnly(text, 'text'),
> > + useKeyOnly(left, 'left aligned'),
> > + useKeyOnly(center, 'center aligned'),
> > + useKeyOnly(right, 'right aligned'),
> > + useKeyOnly(justified, 'justified'),
> > + useKeyOnly(fluid, 'fluid'),
> > + 'container',
> > + className
> > + )
> > +
> > + const ContainerComponent = 'div'
> > + const rest = getUnhandledProps(Container, props)
> > +
> > + return (
> > + <ContainerComponent className={classes} {...rest}>
>
> Since the |ContainerComponent| never changes, let's just use the |<div 
> ...| inline here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/TechnologyAdvice/stardust/pull/277/files/f420da9c4eb10853132c436ba6a06faa82ffeba2#r68423440>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AAWHs1bthM95zj3FbrWdWrv0AFmfQBUbks5qPASrgaJpZM4I9hiP>.
>

-- 
Sent from Postbox 
<https://www.postbox-inc.com/?utm_source=email&utm_medium=siglink&utm_campaign=reach>